README
taylor.fish fork of bsd-games
-----------------------------

The version of `cribbage` in this repository contains the following
modifications:

* Fixed errors in pre-calculated score tables, and added a program to
  generate these tables.
* Fixed additional bugs affecting which cards the computer chooses to
  discard.
* If the environment variable `REALISTICSHUFFLE` is non-empty, the
  computer will simulate overhand and riffle shuffling instead of
  completely randomizing the deck.
* More detail is stored in `criblog` (game time and skunk info).
* If the environment variable `CRIBDETAIL` is set to the path of a
  file, verbose game information will be appended to that file
  (creating it if necessary).

The modifications in this repository are licensed under the same
license as the original (the 3-clause BSD license), except for the
files in `cribbage/generate-score-tables`, which constitute a separate
program licensed under version 3 or later of the GNU General Public
license.

----------------------------------------------------------------------

README for bsd-games
====================

This is the bsd-games package for Linux (and GNU Hurd), containing
ports of all the games from NetBSD-current that are free in the usual
(DFSG/OSD) sense (i.e. modified and unmodified versions can be freely
distributed, including for profit).  NetBSD-current contains two other
games, rogue and larn.  Rogue has a standard BSD licence, but with an
additional contradictory licence that does not allow for-profit
distribution; larn has no licence at all, only a notice that "Copying
for Profit is Prohibited".  If you want rogue, get the
bsd-games-non-free package (which should be on sunsite, but not
tsx-11), and read the README.non-free that it contains: rogue is being
distributed separately, and larn not distributed at all in this Linux
port, in the absence of definite confirmation of what licences really
apply to rogue and larn.  A game not from NetBSD, paranoia, was in
earlier versions of bsd-games-non-free (and bsd-games 1.2 and 1.3) but
has been removed because of the lack of a clear licence.

The games are mostly not enhanced from the NetBSD versions, but many
bugs are fixed.

For installation instructions and details of prerequisites, see the
file "INSTALL".

The statement of year 2000 issues associated with bsd-games and
bsd-games-non-free is in the file "YEAR2000".

The copyrights and licence terms for the games, as best as I can
determine them, are in "COPYING".

This package contains cryptographic software (caesar and rot13).  In
some jurisdictions, use or distribution of these utilities may be
restricted, whether under export control regulations or because these
cryptosystems have been used as technical protection measures to
restrict access to and copying of copyright works.

This package contains the following games:

adventure:	the original adventure by Crowther and Woods
arithmetic:	arithmetic quiz/speed test
atc:		air traffic control
backgammon:	backgammon
banner:		display a message in big letters
battlestar:	adventure game on a battlestar
bcd:		outputs text in an antique form
boggle:		boggle
caesar:		reads fortunes from the game fortune, also some internet posts
canfield:	curses-based solitaire
countmail:	tell you how much new mail you have
cribbage:	cribbage
dab:		dots and boxes
dm:		dungeon master, regulates games playing
factor:		factor a number
fish:		go fish
fortune:	displays a random silly message
gomoku:		gomoku
hack:		exploring the Dungeons of Doom
hangman:	guess the word before it is too late
hunt:		hunt each other in a maze (multiplayer -- great)
mille:		mille borne against the computer
monop:		monopoly
morse:		output morse code
number:		output the English text for a number
phantasia:	interterminal fantasy game
pig:		output text in Pig Latin
pom:		display the phase of the moon
ppt:		outputs text in another antique form
primes:		generate primes
quiz:		random knowledge tests
rain:		attempts to create a rain drop effect (best at 9600 baud)
random:		random lines from a file or random numbers
robots:		well... avoid the robots
sail:		sail your ship into battle
snake:		grab the cash and avoid the snake and exit
tetris:		tetris
trek:		We come in peace, shoot to kill.  It's worse than that, he's
		dead Jim.  Ye cannot change the laws of physics.  It's life
		Jim, but not as we know it.  There's Klingons on the starboard
		bow ...
wargames:	would you like to play a game?
worm:		eat the numbers without running into anything
worms:		random worms scurrying across your screen
wtf:		translate acronyms, e.g. "wtf is WTF"
wump:		hunt the wumpus

If you have questions about bsd-games or bsd-games-non-free you can
contact me at <jsm@polyomino.org.uk>.  Bug reports should state the
versions of ncurses and libc used, unless clearly not relevant (e.g.,
bugs in data files, or where you have an analysis and patch).  If you
want to be informed by email of new releases when they come out,
please let me know.

Known bugs are listed in the file BUGS; please do not send bug reports
for bugs listed there, but patches that fix them are welcome.

For patches, I prefer unidiffs (use "diff -u OLD NEW", or "diff -ruN
OLD-DIRECTORY NEW-DIRECTORY"), but if your diff cannot produce them
then context diffs are OK.  Please remember to include details of the
bug and your analysis of the problem if you are sending a patch; I
have ported the code from NetBSD to Linux, but may not be familiar
with the workings of any particular program, whereas you will be if
you have been tracing and fixing a bug.  The GCC info manual contains
a discussion of how to produce good bug reports.

Notes on specific games:

boggle: by default this game compiles with certain Linux improvements;
if you want the original BSD behaviour, remove -DNEW_STYLE from the
boggle_boggle_DEFS in boggle/boggle/Makefrag.  Defining NEW_STYLE will
provide a more intuitive way of quitting or playing a new game after
time has run out.

dm: the "dungeon master", a program that allows you to control when
users can play games.  (Note that this does not control any private
copies of games they may have.)  I do not claim that this program is
actually useful, but it is included in bsd-games since it is in
NetBSD.  Add -DLOG to the DEFS in dm/Makefrag if you want logging of
games playing (not regularly tested).

fortune: Another enhanced version is available from
<ftp://ibiblio.org/pub/Linux/games/amusements/fortune/fortune-mod-9708.tar.gz>.
unstr is not installed by default.  If you want to install it,
uncomment the relevant lines (installation command and definition of
fortune_unstr_DIRS) in fortune/unstr/Makefrag before doing the top
level "make install".

hunt: may well be unreliable or broken in various ways.  There is some
local configuration possible in hunt/Makeconfig but changing this may
well break things.

Joseph S. Myers
jsm@polyomino.org.uk



Local Variables:
mode: text
End:


debian/changelog
bsdgames (2.17-29) unstable; urgency=medium

  * Team upload.
  * Specify format string literals for all printw() calls.
    Closes: #995597.

 -- Stephen Kitt <skitt@debian.org>  Mon, 11 Oct 2021 14:23:15 +0200

bsdgames (2.17-28) unstable; urgency=medium

  * hack: Fix segv when 'i' (inventory) done after load from saved game.
    Not all name information was saved or recovered leaving pointers to
    nothing.
    Thanks to Richard Henschel <rhenschel@accesscomm.ca>
  * Make choice of c compiler more flexible to help cross building.
    Thanks to Helmut Grohne <helmut@subdivi.de> (Closes: #897048)

 -- Dr. Tobias Quathamer <toddy@debian.org>  Sat, 28 Apr 2018 15:14:43 +0200

bsdgames (2.17-27) unstable; urgency=medium

  * Use debhelper v11
  * Run wrap-and-sort
  * Switch d/watch to version 4
  * Switch Vcs-URLs to salsa.d.o
  * Update Standards-Version to 4.1.4, no changes needed
  * Convert patch system from quilt to git-buildpackage
  * Rename and renumber patches, no content changes
  * Update acronyms from NetBSD. (Closes: #798185)
  * Update primes to NetBSD CVS from 2018-02-03. (Closes: #762653)
  * Better explanation how to read the documentation.
    Thanks to Ben Wong <galadriel@wongs.net> (Closes: #874680)
  * Create and install pdf file for trek documentation

 -- Dr. Tobias Quathamer <toddy@debian.org>  Fri, 27 Apr 2018 11:39:09 +0200

bsdgames (2.17-26) unstable; urgency=medium

  * Fix FTBFS with libncurses6.
    Thanks to Sven Joachim <svenjoac@gmx.de> (Closes: #880492)
  * Update Standards-Version to 4.1.1, no changes needed

 -- Dr. Tobias Quathamer <toddy@debian.org>  Wed, 01 Nov 2017 17:37:54 +0100

bsdgames (2.17-25) unstable; urgency=medium

  * Add libfl-dev to Build-Depends.
    Thanks to Helmut Grohne <helmut@subdivi.de> (Closes: #846412)
  * Use debhelper v10
  * Update to Standards-Version 3.9.8, no changes needed

 -- Dr. Tobias Quathamer <toddy@debian.org>  Mon, 05 Dec 2016 18:27:24 +0100

bsdgames (2.17-24) unstable; urgency=medium

  [ Ben Finney ]
  * adventure:
    - Declare “Provides: adventure” as this is one possible implementation
      of the Colossal Cave Adventure.
    - Use Debian alternatives system to present ‘bsdgames-adventure’ as an
      implementation of ‘adventure’.

  [ Dr. Tobias Quathamer ]
  * Rename adventure to bsdgames-adventure. Together with the changes
    from Ben Finney above, this enables the update-alternatives system.
    Closes: #802422
  * Remove obsolete upstream site from d/watch
  * Update to Standards-Version 3.9.7, no changes needed
  * Use secure Vcs-* URLs

 -- Dr. Tobias Quathamer <toddy@debian.org>  Tue, 05 Apr 2016 10:42:49 +0200

bsdgames (2.17-23) unstable; urgency=medium

  * Update to Standards-Version 3.9.6, no changes needed

 -- Dr. Tobias Quathamer <toddy@debian.org>  Wed, 13 Jan 2016 17:38:52 +0100

bsdgames (2.17-22) unstable; urgency=medium

  * Update Vcs-URLs
  * Update Standards-Version to 3.9.5, no changes needed

 -- Tobias Quathamer <toddy@debian.org>  Tue, 08 Jul 2014 21:25:35 +0200

bsdgames (2.17-21) unstable; urgency=low

  * Switch to debhelper v9
  * Use all hardening features in DEB_BUILD_MAINT_OPTIONS

 -- Tobias Quathamer <toddy@debian.org>  Mon, 26 Mar 2012 11:52:07 +0200

bsdgames (2.17-20) unstable; urgency=low

  * Set maintainer to Debian Games Team, add myself as uploader
  * Add git repository to debian/control
  * rain: Update default delay. Closes: #579062
  * quiz: Add answers to adults/babies. Closes: #659528
  * primes: Document maximum number in manpage. Closes: #636569
  * primes: Updated to NetBSD CVS as of 2012-02-16
  * wtf: Add NMU. Closes: #574036
  * hangman: Error out on 1000 unsuitable words. Closes: #610270
  * Rename files in debian/patches, managed now with gbp-pq
  * Remove Conflicts/Replaces on bsdgames-nonfree (<< 2.14) from
    debian/control, that version is long gone now
  * Remove Build-Depends on dpkg-dev (>= 1.9.0)
  * Use set -e in postinst, postrm, and preinst
  * Fix some typos spotted by Lintian
  * Update to Standards-Version 3.9.3

 -- Tobias Quathamer <toddy@debian.org>  Sat, 25 Feb 2012 17:11:48 +0100

bsdgames (2.17-19) unstable; urgency=low

  * debian/patches/define-dead.diff:
    - Define __dead in include/sys/cdefs.h
  * debian/patches/wump-update.diff:
    - Update wump sources to NetBSD CVS as of 2010-01-14. This fixes
      several bugs, including the over-crowded cave. Thanks to
      Jörg Neulist for the bug report and patch. Closes: #564891

 -- Tobias Quathamer <toddy@debian.org>  Fri, 15 Jan 2010 23:31:41 +0100

bsdgames (2.17-18) unstable; urgency=low

  * debian/patches/sort-acronyms.comp.diff:
    - Sort acronyms.comp with LC_ALL=C. Thanks to Jonas Kölker for the
      patch. Closes: #367689
    - Add JFGI to acronyms.comp. Thanks to Muthu for the suggestion.
      Closes: #475044
  * debian/patches/refresh-robots-screen.diff:
    - Properly refresh screen of robots game with CTRL-L. Thanks to
      tdenniston and Jonathan Davies for the patch and bug report.
      Closes: #562523, LP: #465530
  * debian/patches/anne-boleyn.diff:
    - Fix spelling error of Anne Boleyn in murders quizfile. Thanks
      to Dean Menezes for the bug report. Closes: #562231
  * debian/patches/capitals.diff:
    - Update african capitals and countries in quiz game. Thanks to
      Ulf Schaper for the bug report. Closes: #296027
    - Update american capitals and countries in quiz game.
    - Update asian capitals and countries in quiz game.
    - Update european capitals and countries in quiz game.

 -- Tobias Quathamer <toddy@debian.org>  Sat, 02 Jan 2010 21:59:41 +0100

bsdgames (2.17-17) unstable; urgency=low

  * New maintainer. Closes: #487466
  * Switch to dpkg source format 3.0 (quilt)
  * debian/control:
    - Little changes to the description
    - Bump Standards-Version to 3.8.3
    - Remove Git-Vcs line as it isn't up to date
  * debian/patches/replace-getline.diff:
    - Fix FTBFS by renaming the function getline() to get_line().

trek/setup.c
/*	$NetBSD: setup.c,v 1.8 2003/08/07 09:37:54 agc Exp $	*/

/*
 * Copyright (c) 1980, 1993
 *	The Regents of the University of California.  All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 * 1. Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 * 3. Neither the name of the University nor the names of its contributors
 *    may be used to endorse or promote products derived from this software
 *    without specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS ``AS IS'' AND
 * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
 * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
 * ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
 * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
 * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
 * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
 * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
 * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
 * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 * SUCH DAMAGE.
 */

#include <sys/cdefs.h>
#ifndef lint
#if 0
static char sccsid[] = "@(#)setup.c	8.1 (Berkeley) 5/31/93";
#else
__RCSID("$NetBSD: setup.c,v 1.8 2003/08/07 09:37:54 agc Exp $");
#endif
#endif /* not lint */

#include <stdio.h>
#include <math.h>
#include <string.h>
#include <unistd.h>
#include <stdlib.h>
#include <err.h>
#include "trek.h"
#include "getpar.h"

/*
**  INITIALIZE THE GAME
**
**	The length, skill, and password are read, and the game
**	is initialized.  It is far too difficult to describe all
**	that goes on in here, but it is all straight-line code;
**	give it a look.
**
**	Game restart and tournament games are handled here.
*/

const struct cvntab	Lentab[] =
{
	{ "s",		"hort",		(cmdfun)1,	0 },
	{ "m",		"edium",	(cmdfun)2,	0 },
	{ "l",		"ong",		(cmdfun)4,	0 },
	{ "restart",	"",		(cmdfun)0,	0 },
	{ NULL,		NULL,		NULL,		0 }
};

const struct cvntab	Skitab[] =
{
	{ "n",		"ovice",	(cmdfun)1,	0 },
	{ "f",		"air",		(cmdfun)2,	0 },
	{ "g",		"ood",		(cmdfun)3,	0 },
	{ "e",		"xpert",	(cmdfun)4,	0 },
	{ "c",		"ommodore",	(cmdfun)5,	0 },
	{ "i",		"mpossible",	(cmdfun)6,	0 },
	{ NULL,		NULL,		NULL,		0 }
};

void
setup()
{
	const struct cvntab		*r;
	int		i, j;
	double			f;
	int			d;
	int			klump;
	int			ix, iy;
	struct quad	*q;
	struct event		*e;

	while (1)
	{
		r = getcodpar("What length game", Lentab);
		Game.length = (long) r->value;
		if (Game.length == 0)
		{
			if (restartgame())
				continue;
			return;
		}
		break;
	}
	r = getcodpar("What skill game", Skitab);
	Game.skill = (long) r->value;
	Game.tourn = 0;
	getstrpar("Enter a password", Game.passwd, 14, 0);
	if (strcmp(Game.passwd, "tournament") == 0)
	{
		getstrpar("Enter tournament code", Game.passwd, 14, 0);
		Game.tourn = 1;
		d = 0;
		for (i = 0; Game.passwd[i]; i++)
			d += Game.passwd[i] << i;
		srand(d);
	}
	Param.bases = Now.bases = ranf(6 - Game.skill) + 2;
	if (Game.skill == 6)
		Param.bases = Now.bases = 1;
	Param.time = Now.time = 6.0 * Game.length + 2.0;
	i = Game.skill;
	j = Game.length;
	Param.klings = Now.klings = i * j * 3.5 * (franf() + 0.75);
	if (Param.klings < i * j * 5)
		Param.klings = Now.klings = i * j * 5;
	if (Param.klings <= i)		/* numerical overflow problems */
		Param.klings = Now.klings = 127;
	Param.energy = Ship.energy = 5000;
	Param.torped = Ship.torped = 10;
	Ship.ship = ENTERPRISE;
	Ship.shipname = "Enterprise";
	Param.shield = Ship.shield = 1500;
	Param.resource = Now.resource = Param.klings * Param.time;
	Param.reserves = Ship.reserves = (6 - Game.skill) * 2.0;
	Param.crew = Ship.crew = 387;
	Param.brigfree = Ship.brigfree = 400;
	Ship.shldup = 1;
	Ship.cond = GREEN;
	Ship.warp = 5.0;
	Ship.warp2 = 25.0;
	Ship.warp3 = 125.0;
	Ship.sinsbad = 0;
	Ship.cloaked = 0;
	Param.date = Now.date = (ranf(20) + 20) * 100;
	f = Game.skill;
	f = log(f + 0.5);
	for (i = 0; i < NDEV; i++)
		if (Device[i].name[0] == '*')
			Param.damfac[i] = 0;
		else
			Param.damfac[i] = f;
	/* these probabilities must sum to 1000 */
	Param.damprob[WARP] = 70;	/* warp drive		 7.0% */
	Param.damprob[SRSCAN] = 110;	/* short range scanners	11.0% */
	Param.damprob[LRSCAN] = 110;	/* long range scanners	11.0% */
	Param.damprob[PHASER] = 125;	/* phasers		12.5% */
	Param.damprob[TORPED] = 125;	/* photon torpedoes	12.5% */
	Param.damprob[IMPULSE] = 75;	/* impulse engines	 7.5% */
	Param.damprob[SHIELD] = 150;	/* shield control	15.0% */
	Param.damprob[COMPUTER] = 20;	/* computer		 2.0% */
	Param.damprob[SSRADIO] = 35;	/* subspace radio	 3.5% */
	Param.damprob[LIFESUP] = 30;	/* life support		 3.0% */
	Param.damprob[SINS] = 20;	/* navigation system	 2.0% */
	Param.damprob[CLOAK] = 50;	/* cloaking device	 5.0% */
	Param.damprob[XPORTER] = 80;	/* transporter		 8.0% */
	/* check to see that I didn't blow it */
	for (i = j = 0; i < NDEV; i++)
		j += Param.damprob[i];
	if (j != 1000)
		errx(1, "Device probabilities sum to %d", j);
	Param.dockfac = 0.5;
	Param.regenfac = (5 - Game.skill) * 0.05;
	if (Param.regenfac < 0.0)
		Param.regenfac = 0.0;
	Param.warptime = 10;
	Param.stopengy = 50;
	Param.shupengy = 40;
	i = Game.skill;
	Param.klingpwr = 100 + 150 * i;
	if (i >= 6)
		Param.klingpwr += 150;
	Param.phasfac = 0.8;
	Param.hitfac = 0.5;
	Param.klingcrew = 200;
	Param.srndrprob = 0.0035;
	Param.moveprob[KM_OB] = 45;
	Param.movefac[KM_OB] = .09;
	Param.moveprob[KM_OA] = 40;
	Param.movefac[KM_OA] = -0.05;
	Param.moveprob[KM_EB] = 40;
	Param.movefac[KM_EB] = 0.075;
	Param.moveprob[KM_EA] = 25 + 5 * Game.skill;
	Param.movefac[KM_EA] = -0.06 * Game.skill;
	Param.moveprob[KM_LB] = 0;
	Param.movefac[KM_LB] = 0.0;
	Param.moveprob[KM_LA] = 10 + 10 * Game.skill;
	Param.movefac[KM_LA] = 0.25;
	Param.eventdly[E_SNOVA] = 0.5;
	Param.eventdly[E_LRTB] = 25.0;
	Param.eventdly[E_KATSB] = 1.0;
	Param.eventdly[E_KDESB] = 3.0;
	Param.eventdly[E_ISSUE] = 1.0;
	Param.eventdly[E_SNAP] = 0.5;
	Param.eventdly[E_ENSLV] = 0.5;
	Param.eventdly[E_REPRO] = 2.0;
	Param.navigcrud[0] = 1.50;
	Param.navigcrud[1] = 0.75;
	Param.cloakenergy = 1000;
	Param.energylow = 1000;
	for (i = 0; i < MAXEVENTS; i++)
	{
		e = &Event[i];
		e->date = 1e50;
		e->evcode = 0;
	}
	xsched(E_SNOVA, 1, 0, 0, 0);
	xsched(E_LRTB, Param.klings, 0, 0, 0);
	xsched(E_KATSB, 1, 0, 0, 0);
	xsched(E_ISSUE, 1, 0, 0, 0);
	xsched(E_SNAP, 1, 0, 0, 0);
	Ship.sectx = ranf(NSECTS);
	Ship.secty = ranf(NSECTS);
	Game.killk = Game.kills = Game.killb = 0;
	Game.deaths = Game.negenbar = 0;
	Game.captives = 0;
	Game.killinhab = 0;
	Game.helps = 0;
	Game.killed = 0;
	Game.snap = 0;
	Move.endgame = 0;

	/* setup stars */
	for (i = 0; i < NQUADS; i++)
		for (j = 0; j < NQUADS; j++)
		{
			signed char tmp;
			q = &Quad[i][j];
			q->klings = q->bases = 0;
			q->scanned = -1;
			q->stars = ranf(9) + 1;
			tmp = ranf(3) - q->stars / 5;
			tmp = tmp < 0 ? 0 : tmp;
			q->holes = tmp;
			q->qsystemname = 0;
		}

	/* select inhabited starsystems */
	for (d = 1; d < NINHAB; d++)
	{
		do
		{
			i = ranf(NQUADS);
			j = ranf(NQUADS);
			q = &Quad[i][j];
		} while (q->qsystemname);
		q->qsystemname = d;
	}

	/* position starbases */
	for (i = 0; i < Param.bases; i++)
	{
		while (1)
		{
			ix = ranf(NQUADS);
			iy = ranf(NQUADS);
			q = &Quad[ix][iy];
			if (q->bases > 0)
				continue;
			break;
		}
		q->bases = 1;
		Now.base[i].x = ix;
		Now.base[i].y = iy;
		q->scanned = 1001;
		/* start the Enterprise near starbase */
		if (i == 0)
		{
			Ship.quadx = ix;
			Ship.quady = iy;
		}
	}

	/* position klingons */
	for (i = Param.klings; i > 0; )
	{
		klump = ranf(4) + 1;
		if (klump > i)
			klump = i;
		while (1)
		{
			ix = ranf(NQUADS);
			iy = ranf(NQUADS);
			q = &Quad[ix][iy];
			if (q->klings + klump > MAXKLQUAD)
				continue;
			q->klings += klump;
			i -= klump;
			break;
		}
	}

	/* initialize this quadrant */
	printf("%d Klingons\n%d starbase", Param.klings, Param.bases);
	if (Param.bases > 1)
		printf("s");
	printf(" at %d,%d", Now.base[0].x, Now.base[0].y);
	for (i = 1; i < Param.bases; i++)
		printf(", %d,%d", Now.base[i].x, Now.base[i].y);
	printf("\nIt takes %d units to kill a Klingon\n", Param.klingpwr);
	Move.free = 0;
	initquad(0);
	srscan(1);
	attack(0);
}
